drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -4121,8 +4148,12 @@ static void vlan_dev_unregister(struct net_device *dev)
 	rswitch_rxdmac_free(dev, priv);
 	rswitch_txdmac_free(dev, priv);
 	napi_disable(&rdev->napi);
+	netif_napi_del(&rdev->napi);
+
+	cleanup_all_routes(rdev);
 
 	list_del(&rdev->list);
+	dev->dev.parent = rdev->vlan_parent;
```

Comment:
We probably also need `kfree(rdev)` on device unregister, subsequent vlan device register/unregister will lead to memory leak.